powerplants/src/components/ChooseLayer.js

```diff
@@ -1,190 +1,83 @@
+
+const wmsOptions = [
+   {
+      title: "toner",
+      image: tonerpic,
+      link: "https://stamen-tiles.a.ssl.fastly.net/toner/{z}/{x}/{y}.png",
+   },
+   {
+      title: "terrain",
+      image: terrainpic,
+      link: "https://stamen-tiles.a.ssl.fastly.net/terrain/{z}/{x}/{y}.jpg",
+   },
+   {
+      title: "watercolor",
+      image: watercolorpic,
+      link: "https://stamen-tiles.a.ssl.fastly.net/watercolor/{z}/{x}/{y}.jpg",
+   },
+];
 
 export default function ChooseLayer() {
-  const [terrain, setTerrain] = useState(true);
-  const [toner, setToner] = useState(false);
-  const [watercolor, setWatercolor] = useState(false);
-  const [extended, setExtended] = useState(false);
+   const [currentIndex, setCurrentIndex] = useState(1);
+   const [showWMS, setShowWMS] = useState(wmsOptions[currentIndex].title);
+   const [disableWms, setDisableWms] = useState(false);
 
-  const terrainClick = () => {
-    setToner(false);
-    setWatercolor(false);
-    setTerrain(!terrain);
-  };
-  const tonerClick = () => {
-    setTerrain(false);
-    setWatercolor(false);
-    setToner(!toner);
-  };
-  const watercolorClick = () => {
-    setToner(false);
-    setTerrain(false);
-    setWatercolor(!watercolor);
-  };
+   const mediaIsMobile = useMediaQuery("(max-width:900px)");
 
-  return (
-     <>
-        <MlWmsLayer
-           url="https://stamen-tiles.a.ssl.fastly.net/toner/{z}/{x}/{y}.png"
-           visible={toner}
-           insertBeforeLayer="Plant_data"
-           urlParameters={{ layers: "" }}
-        />
-        <MlWmsLayer
-           url="https://stamen-tiles.a.ssl.fastly.net/terrain/{z}/{x}/{y}.jpg"
-           visible={terrain}
-           insertBeforeLayer="Plant_data"
-           urlParameters={{ layers: "" }}
-        />
-        <MlWmsLayer
-           url="https://stamen-tiles.a.ssl.fastly.net/watercolor/{z}/{x}/{y}.jpg"
-           visible={watercolor}
-           insertBeforeLayer="Plant_data"
-           urlParameters={{ layers: "" }}
-        />
-        <AppBar
-           position="fixed"
-           sx={{
-              top: "auto",
-              bottom: 33,
-              right: 100,
-              width: extended ? 410 : 180,
-              height: 115,
-              opacity: 0.8,
-           }}
-        >
-           <Grid container>
-              <Grid>
-                 <Button
-                    sx={{ height: 115, paddingRight: 3 }}
-                    onClick={() => setExtended(!extended)}
-                 >
-                    <DragHandleIcon
-                       color="secondary"
-                       sx={{ alignSelf: "center", transform: "rotate(90deg)" }}
-                    />
-                 </Button>
-              </Grid>
+   const wmsSetter = (index) => {
+      if (index === currentIndex) {
+         if (disableWms) {
+            setDisableWms(false);
+         } else {
+            setDisableWms(true);
+         }
+      } else {
+         setCurrentIndex(index);
```

Comment:
3 setState calls in a row to 3 useState variables will trigger 3 rerenderings of this component.
It is better to combine the states into a single useState variable.

e.g. as javascript object:
`
{
  currentIndex: number;
  showWms: string;
  disableWms: boolean;
}
`